integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonRDD.scala

```diff
@@ -31,7 +37,7 @@ import org.apache.carbondata.core.util._
  */
 abstract class CarbonRDD[T: ClassTag](@transient sc: SparkContext,
     @transient private var deps: Seq[Dependency[_]]) extends RDD[T](sc, deps) {
-
+  @transient val hadoopConf: Configuration = sc.hadoopConfiguration
```

Comment:
This hadoopConf should be passed via constructor, so add a new parameter in CarbonRDD constructor: `@transient hadoopConf: Configuration`
And modify all subclass of CarbonRDD to pass this parameter.